src/theory/quantifiers/sygus/sygus_grammar_cons.cpp

```diff
@@ -371,6 +371,38 @@ void CegGrammarConstructor::mkSygusConstantsForType(TypeNode type,
   {
     ops.push_back(nm->mkConst(String("")));
   }
+  else if (type.isArray())
+  {
+    TypeNode constituentType = TypeNode::fromType(
+        static_cast<ArrayType>(type.toType()).getConstituentType());
+    Node firstElem;
```

Comment:
I'm not sure we should add a constant for arrays. The issue is that this constant is not handled by the array decision procedure, so we'd fail the verification step of CEGIS.

Regardless, you can use mkGroundTerm or the TypeEnumerator for this type so that you don't have to do special cases here.

But I think we should just do nothing (until the array theory handles array constants).

Reply:
Indeed, I just emulated what the type enumerator for arrays does. I'll add an issue for adding a constant array here once we solve #1667.